autrainer/core/scripts/preprocess_script.py

```diff
@@ -24,6 +25,90 @@ class PreprocessArgs(PreprocessArgs):
     update_frequency: int
 
 
+def preprocess_main(
+    name: str,
+    dataset: DictConfig,
+    preprocess: DictConfig,
+    num_workers: int,
+    update_frequency: int,
+):
+    import os
+    from pathlib import Path
+
+    from tqdm import tqdm
+
+    from autrainer.datasets.utils import AbstractFileHandler
+    from autrainer.transforms import SmartCompose
+
+    print(f" - {name}")
+    if preprocess is None:
+        print("No preprocessing specified. Skipping...")
+        return
+    # set dataset file handler as output handler
+    # as dataset is already configured
+    # to work with the output of the preprocessing script
+    output_file_handler = autrainer.instantiate_shorthand(
+        config=dataset["file_handler"],
+        instance_of=AbstractFileHandler,
+    )
+    output_file_type = dataset["file_type"]
+
+    # override dataset file handling to work with raw audio
+    dataset["file_handler"] = preprocess["file_handler"]
+    # None allows dataset to work with all audio files
+    dataset["file_type"] = None
+    dataset["seed"] = 0  # ignored
+    dataset["batch_size"] = 8  # ignored
+    dataset.pop("criterion")
+    dataset.pop("transform")
+    features_path = dataset.pop("features_path")
+    if features_path is None:
+        features_path = dataset["path"]
+    features_subdir = dataset["features_subdir"]
+
+    dataset["features_subdir"] = None
+    data = autrainer.instantiate(dataset)
+    # manually disable dataset transforms
+    data.train_transform = None
+    data.dev_transform = None
+    data.test_transform = None
+
+    pipeline = SmartCompose(
+        [autrainer.instantiate_shorthand(t) for t in preprocess["pipeline"]]
+    )
+    for d, n in (
+        (data.train_dataset, "train"),
+        (data.dev_dataset, "dev"),
+        (data.test_dataset, "test"),
+    ):
+        loader = torch.utils.data.DataLoader(
+            dataset=d,
+            shuffle=False,
+            num_workers=num_workers,
```

Comment:
All fixed in 721647e

Reply:
In the CLI wrapper function it is mixed up: `num_workers: int = 1` is still set to 1 by default (should be zero) and the docs for `num_workers` should default to zero instead of the `update_frequency`